### PR TITLE
Update dev script for creating a kind cluster with Tekton installed

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

- provide a more unique name for the cluster
- add the missing Triggers manifest for the core interceptors
- set `KO_DOCKER_REPO` and `KIND_CLUSTER_NAME` so `ko` can push the images directly to the cluster

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by AlanGreene** [comment]
@briangleeson I had completely forgotten this script even existed… We should test it out a bit to make sure it's reliable enough then in a future PR we  can document it in the dev docs as an easy way to get started with the latest releases of the Tekton projects.

**Comment by briangleeson** [comment]
Tried it out there and got as far as installing the dashboard, but hit an error:

```
Installing Tekton Dashboard…
Installing ...
error: error validating "C:/Users/BRIANG~1/AppData/Local/Temp/tekton-dashboard-installer.XqP4gU/manifest.yaml": error validating data: ValidationError(RoleBinding): unknown field "<?xml version='1.0' encoding='UTF-8'?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Details>No such object" in io.k8s.api.rbac.v1.RoleBinding; if you choose to ignore these errors, turn validation off with --validate=false
```

Havent dug into it at all though yet

**Comment by AlanGreene** [comment]
Check the output produced by `./scripts/installer build --log-format console --stream-logs --ingress-url tekton-dashboard.127.0.0.1.nip.io`

That should give you the same manifest it's trying to apply. Hopefully there'll be something obvious in the RoleBinding definitions. My guess is that it's a tooling difference on Windows, maybe a different flag, or slightly different output than we're expecting.

**Comment by briangleeson** [comment]
Output:

```
<?xml version='1.0' encoding='UTF-8'?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Details>No such object: tekton-releases/dashboard/previous/0.8.3/installer-tekton-dashboard-release.yaml</Details></Error>---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  labels:
    app.kubernetes.io/component: dashboard
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-dashboard
  name: tekton-dashboard-pipelines
  namespace: tekton-pipelines
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: tekton-dashboard-pipelines
subjects:
  - kind: ServiceAccount
    name: tekton-dashboard
    namespace: tekton-pipelines
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  labels:
    app.kubernetes.io/component: dashboard
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-dashboard
  name: tekton-dashboard-dashboard
  namespace: tekton-pipelines
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: tekton-dashboard-dashboard
subjects:
  - kind: ServiceAccount
    name: tekton-dashboard
    namespace: tekton-pipelines
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  labels:
    app.kubernetes.io/component: dashboard
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-dashboard
  name: tekton-dashboard-triggers
  namespace: tekton-pipelines
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: tekton-dashboard-triggers
subjects:
  - kind: ServiceAccount
    name: tekton-dashboard
    namespace: tekton-pipelines
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    app.kubernetes.io/component: dashboard
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-dashboard
  name: tekton-dashboard-tenant
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: tekton-dashboard-tenant
subjects:
  - kind: ServiceAccount
    name: tekton-dashboard
    namespace: tekton-pipelines
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    app.kubernetes.io/component: dashboard
    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-dashboard
  name: tekton-dashboard-extensions
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: tekton-dashboard-extensions
subjects:
  - kind: ServiceAccount
    name: tekton-dashboard
    namespace: tekton-pipelines
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: tekton-dashboard
  namespace: tekton-pipelines
spec:
  rules:
  - host: tekton-dashboard.127.0.0.1.nip.io
    http:
      paths:
      - backend:
          serviceName: tekton-dashboard
          servicePort: 9097
```

**Comment by AlanGreene** [comment]
That's odd, where's the reference to `0.8.3` coming from? There is no Dashboard 0.8.3 and I don't see it referenced anywhere in the code. Do you have a `VERSION` environment variable set by any chance or have you passed `--version 0.8.3` to the script?

**Comment by briangleeson** [comment]
> That's odd, where's the reference to `0.8.3` coming from? There is no Dashboard 0.8.3 and I don't see it referenced anywhere in the code. Do you have a `VERSION` environment variable set by any chance (we should probably change how we're getting the version internally in the script) or have you passed `--version 0.8.3` to the script?

Yep, apparently I had `VERSION=0.8.3` env var set in the cmd line. Absolutely no idea where that came from

However before checking that, I altered the command to add `version=latest` and that did the trick
`kubectl wait -n tekton-pipelines --for=condition=ready pod --selector=app.kubernetes.io/part-of=tekton-triggers,app.kubernetes.io/component=controller --timeout=90s --version=latest`

File: scripts/prepare-kind-cluster [hunk 1]
```diff
@@ -39,23 +39,32 @@ EOF
...
 
 install_ingress_nginx() {
+  echo 'Installing ingress controller…'
   kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-0.32.0/deploy/static/provider/kind/deploy.yaml
   sleep 20
   kubectl wait -n ingress-nginx --for=condition=ready pod --selector=app.kubernetes.io/component=controller --timeout=90s
 }
 
 install_pipelines() {
+  echo 'Installing Tekton Pipelines…'
```
[the remark this answers is not in the document]
Comment (AlanGreene) [reply scripts/prepare-kind-cluster]: Updated to use the box banners from our e2e tests, but without converting to uppercase

image

**Review by steveodonovan — APPROVED** [review]
/approve
/lgtm

**Pull request merged**